projects/lib/src/board/westernboard.h

```diff
@@ -76,6 +76,33 @@ class LIB_EXPORT WesternBoard : public Board
 		/*! Movement mask for Rook moves. */
 		static const unsigned RookMovement = 8;
 
+		/*! Types of Pawn moves. */
+		enum StepType: const char
+		{
+			 NoStep = 0,      //!< Cannot move here
+			 FreeStep =  1,   //!< May move if target is empty
+			 CaptureStep = 2  //!< Capture opposing piece only
+			 /* FreeOrCaptureStep = FreeStep|CaptureStep //!< like King or Sergeant*/
+		};
+		/*! Stores a move \a type and a move direction \a file for a Pawn move */
+		struct PawnStep { StepType type; int file; };
+		/*!
+		 * Movement mask for Pawn moves.
+		 * Lists pawn move types and relative files
+		 * Forward left diagonal -1, straight 0, right diagonal 1.
+		 * \sa BerolinaBoard
+		 *
+		 * Default: A Pawn can step straight ahead onto a free square or
+		 * capture diagonally forward. So initialise this
+		 * as { {CaptureStep, -1}, {FreeStep, 0}, {CaptureStep, 1} }
+		 */
+		QList<PawnStep>  m_pawnSteps; 
```

Comment:
The detaching problem still persists in the range-based for loops because the container `m_pawnSteps` is non-const. I suggest changing the container type to `QVarLengthArray<PawnStep>`. Then you don't have to change any of the for loops; it's enough to change this one line. We already use QVarLengthArray a lot in the Board classes because it's lean and efficient.